src/components/fileviewer.js

```diff
@@ -0,0 +1,85 @@
+import React, { Component } from 'react';
+
+import * as FetchAPI from '../fetch_data';
+
+export default class FileViewerContainer extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      appError: undefined,
+      rawFile: undefined,
+    };
+
+    const { revision, path } = this.props;
+    FetchAPI.getRawFile(revision, path)
+      .then(response => {
+        if (response.status !== 200) {
+          console.log('Error status code' + response.status);
+          return;
+        }
+        response.text().then(file => {
+          this.setState({ rawFile: file.split('\n') })
+        });
+      })
+      .catch((error) => {
+        console.error(error);
+        this.setState({
+          appError: 'We did not manage to parse the raw-file correctly.',
+        });
+      });
+  }
+
+
+  render() {
+    // console.log(this.props);
+    const { revision, path } = this.props;
+    return (
+      <div>
+        <FileViewerMeta revision={revision} path={path}/>
+        <FileViewer rawFile={this.state.rawFile} />
+      </div>
+    );
+  }
+}
+
+
+
+/* This viewer renders each line of the file with its line number */
+const FileViewer = ({ rawFile }) => {
+  if (!rawFile) {
+    return( <div>"Waiting for file from the backend"</div> );
+  }
+
+  var id = 0;
+  const rawFileLines = rawFile.map((line) => {
+    id = id + 1;
```

Comment:
you should never need to increment a variable in a loop.  In this case, the function given to map can take multiple parameters `(item, index, all_items)`.  https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map

Example:

    const rawFileLines = rawFile.map((line, id) => {
        return(
            <tr key={id+1}>
            <th>{id}</th>